### PR TITLE
New Role: Hunter

### PR DESCRIPTION
**Pull request by fariparedes**
This role has been tested for about two months now so I'm pretty confident the bugs have been mostly ironed out. Based on the Hunter from Throne of Lies: https://tol.fandom.com/wiki/The_Hunter

The Hunter is a new Crewmate Killing role with the ability to track players and execute them if they do anything suspicious. Unlike the Sheriff, the Hunter does not die if they kill an innocent player, however the Hunter may _**only**_ execute players who have given them probable cause.

The Hunter's Kill button starts disabled, and can only be enabled by using their Stalk ability on a player. The ability lasts 25 seconds by default, and has 5 charges. If at any point while the player is being Stalked, they interact in any way with another player, the Hunter's screen flashes and the player becomes Marked. The Hunter's Kill button is enabled for Marked players _**only**_, and they will show up in the meeting screen in a different color and be visible to ghosts.

Crewmate roles which interact with other players, like the Tracker, Detective, etc. are equally vulnerable to the Hunter as are killing roles, so the Hunter must exercise judgment in which Marked players they choose to kill. They may also use their ability solely for its information-gathering purpose, as they will typically know if Stalked players are suspicious for a kill or not. The Hunter by default cannot report its own kills.

Finally, the Hunter has an ace up its sleeve in the form of their passive ability, Retribution. The _**last**_ player who votes for the Hunter, if they are exiled, will be assassinated before the Hunter dies. In my experience, this is often biased towards affecting impostors, as they often save their votes to see which way a meeting is going, and (if the setting is enabled) to preserve their ability to assassinate. On a final 3 vote, if the impostor votes for the Hunter last, the Hunter will kill the impostor first, winning the game for the crew. If the Hunter's last vote was themselves, Retribution does not activate.

### CODE REVIEW

File: source/Patches/CrewmateRoles/HunterMod/CantReport.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    if (Vector2.Distance(truePosition, component.TruePosition) <= __instance.MaxReportDistance)
+                    {
+                        var matches = Murder.KilledPlayers.FirstOrDefault(x => x.PlayerId == component.ParentId);
+                        if (matches != null && matches.KillerId != PlayerControl.LocalPlayer.PlayerId) { 
+                            if (!PhysicsHelpers.AnythingBetween(__instance.Collider, truePosition, component.TruePosition, Constants.ShipOnlyMask, false)) flag2 = true; 
+                        }
+                    }
+                }
+
+            DestroyableSingleton<HudManager>.Instance.ReportButton.SetActive(flag2);
```
Comment (whichtwix) [inline source/Patches/CrewmateRoles/HunterMod/CantReport.cs:39]: the hunter will still be able to report via keybinds, maybe should save this bool as a field and use it in a prefix for doclick?

Comment (fariparedes) [reply source/Patches/CrewmateRoles/HunterMod/CantReport.cs:39]: Do the keybinds work on buttons that are inactive? I would think that's more of an issue with the keybinds.

Comment (whichtwix) [reply source/Patches/CrewmateRoles/HunterMod/CantReport.cs:39]: actually hmm i think i might have not meant keybinds, not sure if those still work but i meant like clicking the body

Comment (fariparedes) [reply source/Patches/CrewmateRoles/HunterMod/CantReport.cs:39]: I wasn't aware you could click on the body to report it. Maybe that's a mod? The Hunter just uses the Sheriff's report button restriction.

**Comment by MyDragonBreath** [comment]
In testing, merged into dev

**Pull request closed without merging**